src/engine-actions/initiate-swap.js

```diff
@@ -0,0 +1,55 @@
+const { sendPayment } = require('../lnd-actions')
+const { networkAddressFormatter } = require('../utils')
+
+/**
+ * Default fee limit for swap routes.
+ * We expect to route swaps through a fee-less hub.
+ * @todo Make this value dynamic.
+ */
+const DEFAULT_FEE_LIMIT = '0'
+
+/**
+ * Initiates a swap
+ *
+ * @param {string} address     - Payment Channel Network Address for the last hop of this leg of the swap
+ * @param {string} swapHash    - base64 string of the swap hash associated with this swap
+ * @param {string} amount      - Int64 string of the amount of outbound currency in its integer units
+ * @param {number} maxTimeLock - Maximum number of seconds that this swap should take to be finalized
+ * @param {number} finalDelta  - delta to use for the time-lock of the CLTV
+ *                               extended to the final hop (in seconds)
+ * @param {string} feeLimit    - Maximum amount of fees to pay on the route
+ * @returns {Promise<void>} - Promise that resolves when the swap is settled
+ */
+async function initiateSwap (address, swapHash, amount, maxTimeLock, finalDelta, feeLimit = DEFAULT_FEE_LIMIT) {
+  this.logger.info(`Initiating swap for ${swapHash} with ${address}`, { address, swapHash, amount })
+
+  const { publicKey: counterpartyPubKey } = networkAddressFormatter.parse(address)
+
+  // Since `maxTimeLock` is a maximum, we use Math.floor to be conservative
+  const cltvLimit = Math.floor(maxTimeLock / this.secondsPerBlock)
+
+  // We use Math.ceil to be sure that the last node in this leg of the swap accepts the payment.
+  const finalCltvDelta = finalDelta ? Math.ceil(finalDelta / this.secondsPerBlock) : 0
```

Comment:
Is the ternary necessary here?

Reply:
If it's not specified it should use the LND default

Reply:
The JSDoc says it's a required parameter and typescript should enforce it, though maybe we haven't tested JSDoc typescript across projects, would be a good thing to verify at some point.